### PR TITLE
chore(api): use library.yaml lint rules and reformat to match

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
Changes API category lint rules to library.yaml and then changes all the offending code to pass analysis. There is no CI enforcement of that here (want to do in separate PR) but there should be no output from `flutter analyze`.

There is one file I deleted here bc realized was dead code. Otherwise no functional changes, just code formatting. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/api/amplify_api/test/amplify_api_config_test.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import 'package:amplify_api/src/amplify_api_config.dart';
+import 'package:amplify_api/src/util/amplify_api_config.dart.dart';
```
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/api/amplify_api/test/amplify_api_config_test.dart:15]: 🤦 will fix, thanks

File: packages/api/amplify_api/lib/src/graphql/graphql_response_decoder.dart [hunk 2]
```diff
@@ -132,9 +135,9 @@ List<GraphQLResponseError>? _deserializeGraphQLResponseErrors(
     return null;
   }
   return errors
-      .cast<Map>()
-      .map((message) => GraphQLResponseError.fromJson(
-            message.cast<String, dynamic>(),
-          ))
+      .cast<Map<String, dynamic>>()
```
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/api/amplify_api/lib/src/graphql/graphql_response_decoder.dart:138]: Apparently so, analyzer complained about Map with implicit dynamic so changed to this. Makes sense now, just doing the casting in once place. Also realized how similar part is in subscription decoder so made this util public and just called it there.

File: packages/api/amplify_api/lib/src/graphql/graphql_subscription_event.dart [hunk 3]
```diff
@@ -62,26 +70,16 @@ class GraphQLSubscriptionEvent {
...
     final payload = json['payload'] as Map?;
     if (payload != null) {
       rawResponse = GraphQLResponse.raw(
         data: payload['data'] as String?,
         errors: (payload['errors'] as List?)
-            ?.cast<Map>()
-            .map((error) => GraphQLResponseError.fromJson(
-                  error.cast<String, dynamic>(),
-                ))
+            ?.cast<Map<String, dynamic>>()
```
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/api/amplify_api/lib/src/graphql/graphql_subscription_event.dart:81]: replaced to call similar util in response decoder (which has tests for query/mutate) but realizing errors in gql subscription events not really tested. Need to research more how you actually get an event like that.

File: packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart [hunk 4]
```diff
@@ -191,6 +196,7 @@ class WebSocketConnection implements Closeable {
   void _cancel(String subscriptionId) {
     _logger.info('Attempting to cancel Operation $subscriptionId');
     send(WebSocketStopMessage(id: subscriptionId));
+    // ignore: todo
```
Comment (Jordan-Nelson) [inline packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart:199]: I don't believe the TODO should cause the lint check to fail. Not sure if that was why you added this or not.

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/api/amplify_api/lib/src/graphql/graphql_subscription_event.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (ragingsquirrel3) [reply packages/api/amplify_api/lib/src/graphql/graphql_subscription_event.dart]: Actually, I just realized this whole file is dead code leftover from method channel impl 🤦 so deleting.

File: packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart (hunk 4 above)
Comment (ragingsquirrel3) [reply packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart]: you are right, removed this. I must have a workspace setting for this or something bc vscode was complaining but not analyzer.

**Pull request merged**